Feature/m0081 meja 1648 architecture 7 17 0 (#421)
MicroEJ/docs: PlatformDeveloperGuide/architectureChangelog.rst

   -  ``QNX65``: BlackBerry QNX 6.5
   -  ``QNX70``: BlackBerry QNX 7.0

[7.17.0] - 2022-06-13
---------------------

Core Engine
~~~~~~~~~~~

-  Fixed potential premature evaluation timeout when Core Engine is not started at the same time as the device.
-  Fixed potential crash during the call of ``LLMJVM_dump`` when printing information about the Garbage Collector.
-  Added new functions to Low Level API ``LLMJVM_MONITOR_impl.h`` (see :ref:`Advanced-Event-Tracing`):

  
   -  ``void LLMJVM_MONITOR_IMPL_on_invoke_method(void* method)``: called by the MicroEJ Core Engine when an method is invoked.
   -  ``void LLMJVM_MONITOR_IMPL_on_return_method(void* method)``: called by the MicroEJ Core Engine when a method returns.

-  [Cortex-M] - Added support for MCU configuration with unaligned access traps enabled (``UNALIGN_TRP`` bit set in ``CCR`` register).

Foundation Libraries
~~~~~~~~~~~~~~~~~~~~

-  Updated ``KF`` to version ``1.6``:
  
   -  Added ``Kernel.canUninstall()`` method.

Integration
~~~~~~~~~~~

> -  Fixed Architecture compatibility issue with Java 11.

Author: New sentence proposed

> -  Fixed missing default value for ShieldedPlug server port when running it with MMM.

Reviewer: can you put the default value 
```suggestion
-  Fixed missing default value for ShieldedPlug server port when running it with MMM (``xxx``).
```

Author: fixed

> -  Updated Memory Map Scripts for ``com.microej.api.vectorgraphics`` library.

Author: fixed

-  Updated Architecture End User License Agreement to version ``SDK 3.1-A``.

Simulator
~~~~~~~~~

> -  Added class file major version check (<=51). Classes mut be compiler for Java 7 or lower. Set the options property ``S3.DisableClassFileVersionCheck`` to ``false`` to disable this verification.

Reviewer: ```suggestion
-  Added class file major version check (<=51). Classes must be compiled for Java 7 or lower. Set the options property ``S3.DisableClassFileVersionCheck`` to ``false`` to disable this verification.
```

> -  Added native method signature in the stack trace of the ``UnsatisfiedLinkError`` thrown when a native method is missing.

Reviewer: use javadoc links (make a whole pass on changelog page ) https://youtrack.cross/issue/M0078DDG-316

Author: All the references to the classes and methods have been converted to links.

-  Fixed ``A[].class.isAssignableFrom(B[].class)`` returning ``false`` instead of ``true`` when  ``B`` is a subclass of ``A``.
-  Fixed potential "Internal limits reached" error when an ``OutOfMemoryError`` is thrown. 
-  Fixed error "Cannot pin objects anymore" when passing repeatedly immutable objects to a native method.
-  Fixed properties not passed correctly to the mocks when the Virtual Device is executed from a path that contains spaces.
-  [Multi] - Fixed unexpected error when ``kernel.kf`` file is missing and KF library is used: "Please specify a 'kernel.kf' file to enable Kernel & Features semantics."
-  [Multi] - Fixed type ``double[]`` not recognized in ``kernel.api`` file.

SOAR
~~~~

-  Fixed internal error when using a BON constant in an if statement at the end of a ``try`` block.
-  Fixed internal error when a ``try`` block ends with an ``assert`` expression while assertions are disabled.
-  [Multi] - Raise a warning instead of an error when duplicated ``.kf`` files are detected in the Kernel classpath. Usual classpath resolution order is used to load the file (see :ref:`chapter.microej.classpath`).

> -  [Multi] - Fixed SOAR error when building a Feature that uses an array of basetypes that is not explicitly declared in the ``kernle.api`` file of the Kernel.

Reviewer: ```suggestion
-  [Multi] - Fixed SOAR error when building a Feature that uses an array of basetypes that is not explicitly declared in the ``kernel.api`` file of the Kernel.
```
